tests/unit/test_asymencryption.cpp

```diff
@@ -0,0 +1,1004 @@
+        "PV+atCNhiPZi7Ts=\n"
+        "-----END PRIVATE KEY-----\n"
+    }
+};
+
+/// \brief RSA 2048-bit Pair public/private key
+const AsymmetricEncryptionTest::KeyPair AsymmetricEncryptionTest::keyPairs2048bit
+{
+    {
+        "-----BEGIN PUBLIC KEY-----\n"
+        "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAotPqMar+G/+YhO7wTKbe\n"
+        "iA6HuYgNIc+4uYwpBbHV+4vZUQnauVUe7ey5YA9qPtyJMNbEmnhEJi5ZwMowyeOP\n"
+        "s0zaxuPNJLMhd5TS9/nfWQ1nzdEchGRiTGrT7EHDzCMm/Ebk/iDaHZqR0roY1s/V\n"
+        "udwyVMEaGfQsTRAVPMPvJztkMENy9k663ou0Pq/svspEditSdEumPHLQER2FQVX6\n"
+        "4T7YuWS6MkR9KKzZPZn70+2CTcaS5E0E5At4b8AzSbvhbR4FsnSKwsRiHH1BE0Yu\n"
+        "xjrVfYzxh/En6FRxolwr/0g8BCx1mnm3OXC7yT3FTFqVtm9RZw8Fr+UpZtIXsk7T\n"
+        "KQIDAQAB\n"
+        "-----END PUBLIC KEY-----\n"
+    },
+    {
+        "-----BEGIN PRIVATE KEY-----\n"
+        "MIIEvAIBADANBgkqhkiG9w0BAQEFAASCBKYwggSiAgEAAoIBAQCi0+oxqv4b/5iE\n"
+        "7vBMpt6IDoe5iA0hz7i5jCkFsdX7i9lRCdq5VR7t7LlgD2o+3Ikw1sSaeEQmLlnA\n"
+        "yjDJ44+zTNrG480ksyF3lNL3+d9ZDWfN0RyEZGJMatPsQcPMIyb8RuT+INodmpHS\n"
+        "uhjWz9W53DJUwRoZ9CxNEBU8w+8nO2QwQ3L2Trrei7Q+r+y+ykR2K1J0S6Y8ctAR\n"
+        "HYVBVfrhPti5ZLoyRH0orNk9mfvT7YJNxpLkTQTkC3hvwDNJu+FtHgWydIrCxGIc\n"
+        "fUETRi7GOtV9jPGH8SfoVHGiXCv/SDwELHWaebc5cLvJPcVMWpW2b1FnDwWv5Slm\n"
+        "0heyTtMpAgMBAAECggEAQF0obC29eO/WKGYTExm7SWkE+a/wEPd9e8sNxDSB84da\n"
+        "hhAxJ6WgLkivFmvBgcvHZ0jeOnlj5vvXer/x/Tw455VvvcL6qpEsn4pywpdBVIBa\n"
+        "Eh+g+Cx+Ntrbnr54WU9j78FOb64oyh75maT6gJ9zSp7iPmJ+oNfGU0gSPQzukkDk\n"
+        "Vf9N118EMh8KUe35pT01zhO4dL9mf/vKrkY/W3r4hEhu66GUH5aI+wbh7zGcCW7L\n"
+        "RWsZcA3xA7zTzI7Tkoe7cTrxOdU3xei2XDUcCFB0bgMcm8Me9GlVAAvI+8BvntCO\n"
+        "KsIi/ennQeMrwIgyoUl7xVGzewnTQ+Vrz94wNfKNAQKBgQDUaFYPJ2TSMg0TnBAE\n"
+        "LUm/poU8e6a80CqheJrnOry6d7spbgTqxWiaAp3stnsmEHRLBmqhrLH3g6B8X2Pl\n"
+        "FM3cMcJir82qUh37vjaWYEPKrJ5pbn2rAMQYyXau7rYICIMW0B9XrlKS9sWS7E3v\n"
+        "4dV/7Sz3CfEpl/w/G0UjzC1SsQKBgQDEPrYIfjW1CiSUVSXU+HFoscA6XxnXdrvu\n"
+        "im3jhJHSCfeMO54bqYwELaERRJSTSalGQfCyFJIWtVMYRdraAJALRSMgrjJpqyBM\n"
+        "0wurapwgPxZv+7GgsecFqO2Z1bo6Bxfd4fgmZo+P0R8d/vNmQAHDo9oDHoMMFSLY\n"
+        "frgOhPb1+QKBgCPWCpb57X+tJBmn3efmgCfOf3Ip5oK6mEav42qZtVe8dZ5Syo+V\n"
+        "iMVbJavXScWtRwGVbo4iZdA9q+jv6xmuPeEloAcoZAmVWyuq9ui+CxH5t6xQEPG1\n"
+        "6l/INbbj3WBTl1JVAMS5S4MyETltluSe9Hdl240+Zyq+UreO8UZN32OxAoGAS7nL\n"
+        "MmHJRa/BR7/HlnO8B2M3ww55gUwe3sBqAPZh8oPIE0PaEeCN1/rngStOjbtJ54OA\n"
+        "ZqdLbBnSE7a46ZB4BPFLycuORnozP5HcmevMjSDq/pFIDLM5a1SxQsBXWRbtncAZ\n"
+        "qm+/DAbtE/0SXS1rwNuUP8nhVaM+S/4x9eBXD7kCgYAQmsS/L5ZZEDTVtqJhiyGj\n"
+        "tnpq6kJKVSsa3WItN47HCE0nuPpjtH3Df/w8CQg2AUA/IjvYLcpTYgJlYt9mm+bF\n"
+        "ytMgYE9OF7bDJzqfAqolINa5KQIXZhmdE5vdDWogxZzy0SsaYCo/+0pR3Q0PCuQa\n"
+        "UwMtJkDrJSjst498ZuyGpg==\n"
+        "-----END PRIVATE KEY-----\n"
+    }
+};
+
+/// \brief Data set used to test successful (nominal) use-cases, both encryption and decryption
```

Comment:
Value-Parameterized Tests from gtest might be a more elegant solution to performing the same test with different inputs.